rcgen/src/lib.rs

```diff
@@ -1710,7 +1712,10 @@ impl Certificate {
 	/// should not call `serialize_request_der` and then `serialize_request_pem`. This will
 	/// result in two different CSRs. Instead call only `serialize_request_pem` and base64
 	/// decode the inner content to get the DER encoded CSR using a library like `pem`.
-	pub fn serialize_request_der(&self, subject_key: &KeyPair) -> Result<Vec<u8>, Error> {
+	pub fn serialize_request_der(
+		&self,
+		subject_key: &KeyPair,
+	) -> Result<CertificateSigningRequestDer<'static>, Error> {
```

Comment:
Could you also mention in the docs how this can be converted into a vec or byte slice so that it can be saved into a file, ideally with an example that we test or rustdoc links so that we notice the breakage.